src/common.ts

```diff
@@ -238,11 +244,14 @@ export type UpdateGatewaySettingsParamsBase = {
   properties?: string;
   protocol?: AllowedProtocols;
   autoStake?: boolean;
+  observerWallet?: WalletAddress;
 };
 
 // Utility type to require at least one of the fields
-export type AtLeastOne<T, U = { [K in keyof T]-?: T[K] }> = Partial<U> &
-  { [K in keyof U]: Required<Pick<U, K>> }[keyof U];
+export type AtLeastOne<
```

Comment:
I HAD checked this before but not down enough avenues - this now properly accepts multiple properties as well as a single property, requiring them to be non nullable, and throws on empty objects.